lib/ansible/modules/crypto/openssl_privatekey.py

```diff
@@ -157,9 +159,7 @@ def __init__(self, module):
         self.privatekey = None
         self.fingerprint = {}
 
-        self.mode = module.params['mode']
-        if not self.mode:
-            self.mode = int('0600', 8)
+        self.mode = module.params.get('mode', None)
```

Comment:
Just so you know, because of how param parsing works, this is equivalent to:
``` python
self.mode = module.params['mode']
```
Explicit can be better than implicit though, so I don't object to this.

Do we want to keep specifying the default mode here?  I would leave it here if we do not care that the user did not specify a mode.

Note that the former check is incorrect because it catches a mode of 0 (no permissions).  Use this instead:
```
self.mode = module.params.get('mode', None)
if self.mode is None:
    self.mode = 0o600
```
(Note that we no longer have to use int() because we no longer support python-2.4)

Reply:
Yes. We need 0o600 as default permission for private key.